bitnami/kafka/templates/NOTES.txt

```diff
@@ -50,13 +53,67 @@ IPs for Kafka brokers. To complete your deployment follow the steps below:
 
 ** Please be patient while the chart is being deployed **
 
-Kafka can be accessed via port {{ .Values.service.port }} on the following DNS name from within your cluster:
+Kafka can be accessed by consumers via port {{ $servicePort }} on the following DNS name from within your cluster:
 
-    {{ template "kafka.fullname" . }}.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
+    {{ $fullname }}.{{ $releaseNamespace }}.svc.{{ $clusterDomain }}
 
-To create a a pod that you can use as a Kafka client run the following command:
+Each Kafka broker can be accessed by producers via port {{ $servicePort }} on the following DNS name(s) from within your cluster:
 
-    kubectl run {{ template "kafka.fullname" . }}-client --rm --tty -i --restart='Never' --image {{ template "kafka.image" . }} --namespace {{ .Release.Namespace }} --command -- bash
+    {{- range $i, $e := until $replicaCount }}
+    {{ $fullname }}-{{ $i }}.{{ $fullname }}-headless.{{ $releaseNamespace }}.svc.{{ $clusterDomain }}
+    {{- end }}
+
+{{- if (include "kafka.client.saslAuthentication" .) }}
+
+You need to configure your Kafka client to access using SASL authentication. You can generate the 'kafka_jaas.conf' and 'client.properties' files using the commands below:
+
+    cat > /tmp/kafka_jaas.conf << EOF
+    KafkaClient {
+        org.apache.kafka.common.security.plain.PlainLoginModule required
+        username="{{ .Values.auth.jaas.clientUser }}"
+        password="$(kubectl get secret {{ $fullname }}-jaas -n {{ $releaseNamespace }} -o jsonpath='{.data.client-password}' | base64 --decode)";
+    };
+    EOF
+    cat > /tmp/client.properties << EOF
+    {{- if eq .Values.auth.clientProtocol "sasl_tls" }}
+    ssl.truststore.location=/tmp/kafka.truststore.jks
+    {{- if .Values.auth.jksPassword }}
+    ssl.truststore.password={{ .Values.auth.jksPassword }}
+    {{- end }}
+    {{- end }}
+    security.protocol={{ $clientProtocol }}
+    sasl.mechanism=PLAIN
+    EOF
```

Comment:
My concern with this "cat" command is that it cannot be copy-pasted, because the EOF should be right at the beginning of a line. Can we find a better way to do this? Even if we ask the users to create a file with the specific contents.

Reply:
Sure, modifying it